spinalcordtoolbox/math.py

```diff
@@ -29,6 +29,19 @@ def dilate(data, size, shape, dim=None):
     # Create structuring element of desired shape and radius
     # Note: the trick here is to use the variable shape as the skimage.morphology function itself
     selem = globals()[shape](size)
+    # If 2d kernel, replicate it along the specified dimension
+    if len(selem.shape) == 2:
+        selem3d = np.zeros([selem.shape[0]]*3)  # Note: selem.shape[0] and selem.shape[1] are supposed to be the same
```

Comment:
This comment could be an `assert` or a `ValueError`.

Reply:
if false, an `assert` will raise an `AssertionError`, but in this case, `selem.shape` could be 2 or 3 (both are fine). So, does assert make sense here? (i.e. 2 or 3 are not "wrong" per se)

Reply:
You could write

```
if not (len(selem.shape) in [2,3] and selem.shape[0] == selem.shape[1]):
    raise ValueError("Invalid shape") 
```

Reply:
And I've come around to the idea that ValueErrors are better than asserts because the latter can be turned off.